src/Action.hs

```diff
@@ -192,14 +219,17 @@ modifyCurrentUser :: (ActionPersist r m, ActionUserHandler m) => (User -> User)
 modifyCurrentUser f =
   currentUser >>= persistent . flip modifyUser f . (^. _Id)
 
+noCurrentUser :: ActionUserHandler m => m Bool
+noCurrentUser = (UserLoggedOut ==) <$> userState
```

Comment:
I would call this `isLoggedIn` and invert its value.  Similarly, when used in `Frontend.Core`, I would remove the negation from `noLoggedInUser`.